packages/react-liveness/src/components/FaceLivenessDetector/service/machine/index.ts

```diff
@@ -198,46 +202,46 @@ export const livenessMachine = createMachine<LivenessContext, LivenessEvent>(
         invoke: {
           src: 'openLivenessStreamConnection',
           onDone: {
-            target: 'waitForSessionInfo',
+            target: 'notRecording',
             actions: [
               'updateLivenessStreamProvider',
               'spawnResponseStreamActor',
             ],
           },
         },
       },
-      waitForSessionInfo: {
-        after: {
-          0: {
-            target: 'notRecording',
-            cond: 'hasServerSessionInfo',
-          },
-          100: { target: 'waitForSessionInfo' },
-        },
-      },
       notRecording: {
         on: {
           START_RECORDING: 'recording', // if countdown completes while face is far enough, start recording
         },
-        initial: 'detectFaceDistanceDuringCountdown',
+        initial: 'waitForSessionInfo',
         states: {
-          detectFaceDistanceDuringCountdown: {
+          waitForSessionInfo: {
+            after: {
+              0: {
+                target: '#livenessMachine.recording',
+                cond: 'hasServerSessionInfo',
+              },
+              10: { target: 'detectFaceDistanceDuringLoading' },
```

Comment:
Just for my curiosity, why do we have to do these two tasks async? (0 vs 10). Is it possible to combine them?

Reply:
I guess this is the pattern I originally saw done in this machine for essentially doing conditionals. If the first condition does not pass then we do the second one 10 ms later. If there is a better way of doing it happy to change it :)

Reply:
If you're only interested in doing one or the other, you can do this:

```ts
waitForSessionInfo: {
  always: [
    {
      target: '#livenessMachine.recording',
      cond: 'hasServerSessionInfo',
    },
    { target: 'detectFaceDistanceDuringLoading' },
  ],
},
```
and essentially work as `if` `else` statement. This is more [idiomatic](https://xstate.js.org/docs/guides/guards.html#multiple-guards) to handle conditional transitions 🙏

Reply:
ah thanks will change!